### ISSUE TITLE
Cannot disable "typescript is not installed" warning as instructed

### ISSUE THREAD

**Issue by anomiex** [gajus/eslint-plugin-jsdoc#1773]
## Expected behavior

The warning issued states

> ⚠️ typescript is not installed. `jsdoc/no-unnecessary-type-assertion` will not work. To disable this warning, you must disable the rule.

If the `jsdoc/no-unnecessary-type-assertion` rule is not enabled, this warning should not be issued.

## Actual behavior

The warning is issued as soon as the package is imported, even if the plugin isn't enabled in the config at all.

## ESLint Config

```js
import jsdoc from 'eslint-plugin-jsdoc';

export default [
    {
        // Normally we'd configure eslint-plugin-jsdoc and such in here, but it's not actually necessary to illustrate the bug.
    },
];
```

Note that you have to make sure that the optional peer dependency on `typescript` isn't automatically installed. For example, with pnpm, set `autoInstallPeers: false` in `pnpm-workspace.yaml`.

## ESLint sample

No specific code is necessary, just run `eslint`.

## Environment

- Node version: v24.21.0
- ESLint version v10.10.0
- `eslint-plugin-jsdoc` version: 64.5.3

## Additional notes

PR #1769 fixed one part of #1768, but not this part. It seemed cleanest to file a separate report. 🙂

I suppose you could instead change the warning to state "To disable this warning, install typescript" if you'd rather.
